app/models/concerns/project_search.rb

```diff
@@ -51,6 +51,13 @@ def exact_name
       name
     end
 
+    def marshal_dump
```

Comment:
I wonder if this means we aren't disabling or mocking something for elasticsearch properly

Reply:
Outside of what's in spec_helper I couldn't find any elasticsearch setup or anything... I think its probably an in memory test version. This link suggests we could maybe be doing more? I don't know much about elasticsearch yet though. https://medium.com/@rowanoulton/testing-elasticsearch-in-rails-22a3296d989

Reply:
Its using an actual elasticsearch instance, I don't think there's any mocking involved.

Reply:
This essentially was a bug in our code here that the upgrade from Rails 5.1.6 -> Rails 5.2.2 exposed. The prior behavior of the NullStore in test was to not try to compress the object being cached at all. In 5.2.2 the behavior changed, thus calling Marshal.dump during the test, and exposing the fact that __elasticsearch__ causes problems in Marshal.dump. Why it only seems to occur on a freshly created object, I'm not entirely sure, but it does explain how this issue surfaced. The relevant pieces of the 5.1.6 cache code is here:
https://github.com/rails/rails/blob/1ad8ac7a1295e74d7499f6e1993d93bb0fd579e4/activesupport/lib/active_support/cache.rb#L670-L679 At least in our configuration, options[:compress], is nil so that this statement is always false. However, in the 5.2.2 version, https://github.com/rails/rails/blob/09ca939eabc8b0cda865ecf2893c48800a7c51e3/activesupport/lib/active_support/cache.rb#L738-L744 the default value for compress is true, which actually attempts to perform the compress method and surfacing this issue.

Reply:
Does that mean that we should change the value to false? As otherwise our cache objects aren't going to be valid with new rails?

Reply:
I believe its not affected for production as we appear to have compression on already for our production config. https://github.com/librariesio/libraries.io/blob/master/config/environments/production.rb#L86-L94